Dungeon.java

```diff
@@ -0,0 +1,267 @@
+/*
+ * Course:        TCSS143B - Fundamentals of Object-Oriented Programming
+ * File Name:     Dungeon.java
+ * Assignment:    5
+ * Due Date:      10 June 2021
+ * Instructor:    Tom Capaul
+ */
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Random;
+import java.util.Set;
+
+/**
+ * Allows the representation of a Dungeon as a 2D array of Room instances.
+ * Annotates all locales of interest within the array and keeps track of
+ * the Hero.
+ *
+ * @author Anthony Nguyen anguyenq@uw.edu
+ * @version 27 May 2021
+ */
+public class Dungeon {
+	
+	/*
+	 * DEBUG MAIN
+	 * Utilized with a placeholder Room class that just returns "?" as its
+	 * toString()
+	 */
+	public static void main(String[] theArgs) {
+		Dungeon test = new Dungeon();
+		System.out.println(test);		
+	}
+	
+	/** 2D array of Rooms within an instance of this class. */
+	private Room[][] myMaze;
+	
+	/** 
+	 * 2D integer array holding "coordinates" of locales of interest within
+	 * an instance of this class.
+	 */
+	private int[][] myLocales;
+	
+	/** Creates an instance of this class. */
+	public Dungeon() {
+		/* 
+		 * Corresponding rows of myLocales
+		 * 0 - Entrance
+		 * 1 - Exit
+		 * 2 - Crown1
+		 * 3 - Crown2
+		 * 4 - Hero (Starts out at the entrance)
+		 */
+		myLocales = setLocales();
+		myMaze = setMaze();
+	}
+	
+	/**

```

Comment:
Just clarifying- this hashmap is keeping track of where the character has already been inside the maze, correct?

Reply:
No, it's keeping track of what coordinates have been used as it determines a potential location for each item of interest (entrance, exit, crown piece, etc.) as they are mutually exclusive.